src/python/pants/backend/jvm/tasks/coverage/base.py

```diff
@@ -106,14 +106,13 @@ class BaseCoverage(CoverageInterface):
 
   @classmethod
   def register_options(cls, register, register_jvm_tool):
+    pass
+
+  @classmethod
+  def register_generic_coverage_options(cls, register, register_jvm_tool):
     register('--coverage', type=bool, fingerprint=True, help='Collect code coverage data.')
 
-    # TODO(John Sirois): Fold this base class up into `Cobertura` once `--coverage-processor` is
-    # killed.
-    register('--coverage-processor', advanced=True, default='cobertura', choices=['cobertura'],
-             removal_version='1.6.0.dev0',
-             removal_hint='Only cobertura is supported for code coverage so this option can be '
-                          'omitted.',
+    register('--coverage-processor', advanced=True, default='cobertura', choices=['cobertura', 'jacoco'],
```

Comment:
Given that this is an abstract base class, defining this option here doesn't make sense to me.

Each instantiation of the base class will have its own instances of the options, so the `CoberturaCoverage` and `JacocoCoverage` subsystems would both have a `--coverage-processor` option, each of which would refer to both systems.